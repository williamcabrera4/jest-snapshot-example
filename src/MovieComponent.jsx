import React from 'react';

const MovieComponent = ({ movie }) => (
  <div>
    <h1>My Movie</h1>
    <ul>
      <li><b>Title:</b> {movie.title}</li>
      <li><b>Description:</b> {movie.description}</li>
      <li><b>Language:</b> {movie.language}</li>
      <li>
        <b>Writers:</b>
        {movie.writers.map((writer) => writer).join(',')}
      </li>
      <li>
        <b>Actors:</b>
        {movie.actors.map((actor) => actor).join(',')}
      </li>
      <li>
        <b>Genres:</b>
        {movie.genres.map((genre) => genre).join(',')}
      </li>
    </ul>
  </div>
);

export default MovieComponent;
