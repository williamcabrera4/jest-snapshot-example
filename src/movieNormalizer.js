const movieNormalizer = (movieData) => ({
  title: movieData.Title,
  description: movieData.Plot,
  language: movieData.Language,
  writers: movieData.Writer.split(','),
  actors: movieData.Actors.split(','),
  genres: movieData.Genre.split(','),
});

export default movieNormalizer;
