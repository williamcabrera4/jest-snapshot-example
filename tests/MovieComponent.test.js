import React from 'react';
import renderer from 'react-test-renderer';

import movie from './movie.json';
import movieNormalizer from '../src/movieNormalizer';
import MovieComponent from '../src/MovieComponent';

describe('Movie Component', () => {

  test('should render the movie information', () => {
    const normalizedMovie = movieNormalizer(movie);
    const component = renderer.create(<MovieComponent movie={normalizedMovie} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});