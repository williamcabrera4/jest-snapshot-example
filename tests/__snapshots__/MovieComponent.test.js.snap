// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component should render the movie information 1`] = `
<div>
  <h1>
    My Movie
  </h1>
  <ul>
    <li>
      <b>
        Title:
      </b>
       
      Game of Thrones
    </li>
    <li>
      <b>
        Description:
      </b>
       
      Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.
    </li>
    <li>
      <b>
        Language:
      </b>
       
      English
    </li>
    <li>
      <b>
        Writers:
      </b>
      David Benioff, D.B. Weiss
    </li>
    <li>
      <b>
        Actors:
      </b>
      Peter Dinklage, Lena Headey, Emilia Clarke, Kit Harington
    </li>
    <li>
      <b>
        Genres:
      </b>
      Adventure, Drama, Fantasy
    </li>
  </ul>
</div>
`;
